lib/archethic/crypto.ex

```diff
@@ -1141,22 +1141,26 @@ defmodule Archethic.Crypto do
   """
   @spec get_root_ca_public_key(key()) :: binary()
   def get_root_ca_public_key(<<curve::8, origin_id::8, _::binary>>) do
-    case Keyword.get(@certification_public_keys, ID.to_origin(origin_id)) do
-      nil ->
-        "no_key"
-
-      # Only for dev
-      [] ->
-        ""
-
-      curves when is_list(curves) ->
-        case Keyword.get(curves, ID.to_curve(curve)) do
-          nil ->
-            "no_key"
-
-          public_key ->
-            public_key
-        end
+    if System.get_env("ARCHETHIC_NETWORK_TYPE") == "testnet" do
+      ""
+    else
+      case Keyword.get(@certification_public_keys, ID.to_origin(origin_id)) do
```

Comment:
Ideally it should be on the conf by replacing this existing conf with the check on the network's type
```elixir
config :archethic, Archethic.Crypto,
  root_ca_public_keys: [
    software: case System.get_env("ARCHETHIC_NETWORK_TYPE") do
      "testnet" -> []
       _ ->
         [
           secp256r1: ...
         ]
    end
...
    ]
  ]
```

So the check could remain constant and without direct access to the system environment variable from the code.

Reply:
ok